### ISSUE TITLE
Generate Usage spec file

### ISSUE THREAD

**Issue by HipsterBrown** [bloomberg/stricli#39]
**Is your feature request related to a problem? Please describe.**
I've really enjoyed using this library, and I think the docs and parameters data can be used for some interesting integrations, like generating more shell auto-completions, manpages, and Markdown documentation. 

**Describe the solution you'd like**
Provide an option or "plugin" (similar to the existing auto-completion package) that generates a [Usage spec](https://usage.jdx.dev/spec/) file to be used by the `usage` CLI for accomplishing the feature requests listed above. 

**Describe alternatives you've considered**
Create individual packages that provide each of these features, including extending support for other shells besides bash in the existing auto-completion package. 

**Additional context**
This is an early spec and may be subject to change, so I understand if it's not ready for adding to this core repo.

**Comment by molisani** [comment]
I don't think we have the desire to support this directly (at least at this time), but it definitely could still be built today. The types for `Application`/`RouteMap`/`Command` already exposed by `@stricli/core` should be enough to construct one of these Usage spec files. This could be built as a separate package, and then potentially it could be incorporated as a first-party package if it made sense for our larger product plan. Please let me know on this issue if there's any data that the API does not provide.

**Comment by molisani** [comment]
I'm going to close this issue out for now as there isn't anything actionable left on our end, but please don't hesitate to reopen or open a new issue if anything comes up with the API we expose.
